### PR TITLE
Update documentation of SphericalManifold::normal_vector.

### PR DESCRIPTION
**Pull request by bangerth**
I needed to explain the issue to myself again after trying to understand https://github.com/geodynamics/aspect/pull/3377.

/rebuild

### CODE REVIEW

File: source/grid/manifold_lib.cc [hunk 1]
```diff
@@ -419,10 +419,17 @@ SphericalManifold<dim, spacedim>::normal_vector(
   const typename Triangulation<dim, spacedim>::face_iterator &face,
   const Point<spacedim> &                                     p) const
 {
-  // if the maximum deviation for the distance from the vertices to the center
-  // is less than 1.e-5 of the minimum distance to the first vertex, assume we
-  // can simply return p-center. otherwise, we compute the normal using
-  // get_normal_vector
+  // Let us first test whether we are on a "horizontal" or "vertical"
```
Comment (gassmoeller) [inline source/grid/manifold_lib.cc]: I would rephrase this. There are other option than "horizontal" or "vertical" and we only test for "horizontal" and treat it separately. Maybe:
```suggestion
  // Let us first test whether we are on a "horizontal" face. If so ...
```

File: source/grid/manifold_lib.cc [hunk 2]
```diff
@@ -419,10 +419,17 @@ SphericalManifold<dim, spacedim>::normal_vector(
...
+  // face. In the former case, the normal vector is easy to compute
+  // since it is proportional to the vector from the center to the
+  // point 'p'.
+  //
+  // We test whether that is the case by checking that the vertices
+  // all have roughly the same distance from the center: If the
+  // maximum deviation for the distances from the vertices to the
+  // center is less than 1.e-10 of the distance between vertices (as
```
Comment (gassmoeller) [inline source/grid/manifold_lib.cc]: Oh also it is 1.e-5. The 1e-10 is comparing the square of the distances.

Comment (bangerth) [reply source/grid/manifold_lib.cc]: Ah yes, good catch -- I took the opportunity to also rename the variables to reflect the fact that they represent *squared* distances.

**Comment by bangerth** [comment]
Updated. I also updated the second place where essentially the same code path and text appeared.
